src/compiler/parser.ts

```diff
@@ -5727,6 +5727,9 @@ namespace ts {
             function parseJSDocParameter(): ParameterDeclaration {
                 const parameter = <ParameterDeclaration>createNode(SyntaxKind.Parameter);
                 parameter.type = parseJSDocType();
+                if (parseOptional(SyntaxKind.EqualsToken)) {
+                    parameter.questionToken = createNode(SyntaxKind.QuestionToken);

```

Comment:
Shouldn't the SyntaxKind match the actual contents of token? Here we are marking a '=' token as a QuestionToken.

I get it works, but seems a little *dirty*. Is there precedence in the code for marking tokens with a SyntaxKind which doesn't match the token contents? If not, doesn't seem like we should start.